[API updates] Fleet Desktop > Self-service: categories & updates (#27681)
fleetdm/fleet: docs/REST API/rest-api.md

| id | integer | path | ID of the software title being updated. |
| software        | file    | form | Installer package file. Supported packages are .pkg, .msi, .exe, .deb, and .rpm.   |
| team_id         | integer | form | **Required**. The team ID. Updates a software package in the specified team. |

> | categories        | string | form | One or more (comma-separated) category name, as shown in the UI. Sets the software package to the specified categories. |

Reviewer: @eugkuo here and for upload package could we change the type to be `array` and remove the bit about the comma-separated list? This will make it consistent with how we implement labels (and make it a bit easier to parse the response in the server code).

Reviewer 2: @jahzielv I was eng DRI on this I think, so answer's yes.

Reviewer: @iansltx cool! I can push that change if no objection

> | categories        | string | form | One or more (comma-separated) categories. Use `categories` to group self-service software on your end users' **Fleet Desktop > My device** page. See supported categories [here](https://fleetdm.com/docs/configuration/yaml-files#example7).|

Reviewer 3: Wait, @eugkuo can users apply multiple categories in the UI to view software that belongs to more than one category?

Author: @RachelElysia No. Users select one category in the UI. If the software belongs to more than one category it shows up in each category it belogs to when the category is selected. So, if a software is added to every category it will show up in every category when it's selected.

> | categories        | string | form | One or more (comma-separated) categories. Use `categories` to group self-service software on your end users' **Fleet Desktop > My device** page. See supported categories [here](https://fleetdm.com/docs/configuration/yaml-files#example7).|

Reviewer 2: ```suggestion
| categories        | string[] | form | One or more of the [supported categories](https://fleetdm.com/docs/configuration/yaml-files#supported-software-categories), used to group self-service software on your end users' **Fleet Desktop > My device** page. |
```

> | install_script  | string | form | Command that Fleet runs to install software. If not specified Fleet runs the [default install command](https://github.com/fleetdm/fleet/tree/f71a1f183cc6736205510580c8366153ea083a8d/pkg/file/scripts) for each package type. |

Reviewer 2: ```suggestion
| install_script  | string | form | Command that Fleet runs to install software. If not specified Fleet runs the [default install command](https://github.com/fleetdm/fleet/tree/main/pkg/file/scripts) for each package type. |
```

| id | integer | path | ID of the software title being updated. |
| software        | file    | form | Installer package file. Supported packages are .pkg, .msi, .exe, .deb, and .rpm.   |
| team_id         | integer | form | **Required**. The team ID. Updates a software package in the specified team. |

> | categories        | string[] | form | One or more of the [supported categories](https://fleetdm.com/docs/configuration/yaml-files#supported-software-categories), used to group self-service software on your end users' **Fleet Desktop > My device** page. |

Reviewer 2: ```suggestion
| categories        | string[] | form | Zero or more of the [supported categories](https://fleetdm.com/docs/configuration/yaml-files#supported-software-categories), used to group self-service software on your end users' **Fleet Desktop > My device** page. Software with no categories will be still be shown under **All**. |
```

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |
| team_id       | integer | body | **Required**. The team ID. Edits App Store apps from the specified team.  |

> | categories | string[] | body | A list of categories for the VPP app, as they're shown in the web UI. Replaces the existing categories list with the specified categories. |

Reviewer 2: ```suggestion
| categories | string[] | body | Zero or more of the [supported categories](https://fleetdm.com/docs/configuration/yaml-files#supported-software-categories), used to group self-service software on your end users' **Fleet Desktop > My device** page. Software with no categories will be still be shown under **All**. |
```
